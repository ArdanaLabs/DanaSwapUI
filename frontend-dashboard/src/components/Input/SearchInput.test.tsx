import Adapter from "@wojtekmaj/enzyme-adapter-react-17"
import * as Enzyme from "enzyme"
import { shallow } from "enzyme"
import React from "react"
import { SearchInput } from "."

Enzyme.configure({ adapter: new Adapter() })

jest.mock("react-redux", () => ({
  useDispatch: () => {},
  useSelector: () => ({
    user: {
      userDarkMode: null,
      mediaDarkMode: false,
      timestamp: "dfd",
    },
  }),
}))

let wrapper: Enzyme.ShallowWrapper
const onFilterChange = jest.fn()

describe("Components / Box / TokenBox", () => {
  beforeEach(() => {
    wrapper = shallow(
      <SearchInput
        className={"someClassName"}
        value={"someValue"}
        placeholder="SEARCH..."
        isIcon={true}
        onChange={onFilterChange}
      />
    )
  })

  it("renders", () => {
    expect(wrapper).toMatchSnapshot()
  })
  it("should contain input tag", () => {
    expect(wrapper.find("input")).toBeTruthy()
  })
  it("should be correct props", () => {
    expect(wrapper.find("input").prop("type")).toEqual("text")
    expect(wrapper.find("input").prop("value")).toEqual("someValue")
    expect(wrapper.find("input").prop("className")).toEqual("someClassName")
    expect(wrapper.find("input").prop("onChange")).toEqual(onFilterChange)
    expect(wrapper.find("input").prop("placeholder")).toEqual("SEARCH...")
  })
  it("changes input value onChange event", () => {
    wrapper.find("input").simulate("change", "My new value")
    expect(onFilterChange).toBeCalledWith("My new value")
  })
})
