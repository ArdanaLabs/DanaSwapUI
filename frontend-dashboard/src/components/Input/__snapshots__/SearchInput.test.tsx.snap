// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Box / TokenBox renders 1`] = `ShallowWrapper {}`;
